### DISCUSSION TITLE
ImportError when rum a SB script in a Docker

### DISCUSSION THREAD

**Discussion by renatopejon** [seleniumbase/SeleniumBase#2681] [Q&A] [answered]
Hi folks,

I have a problem running a script using SB when I build a docker container.

The script runs fine on my local machine, it also runs fine when I go through the interactive docker using "docker run -i -t seleniumbase", then I go inside the folder where the script is, and run with "python myscript.py".

The problem is happening when I'm trying to run it on the container startup. When I try to do it I got the error: "ImportError: No module named seleniumbase"

I tried to do it with Dockerfile like that:
`CMD ["python", "myscript.py"]`

And also building the seleniumbase container and running it like this:
`docker run -it seleniumbase /bin/bash -c 'python myscript.py'`

Here is the myscript.py content:
```python
from seleniumbase import SB

driver = SB(uc=True, xvfb=True, headed=True)

with SB() as sb:
    try:
        sb.driver.get("https://bot.sannysoft.com/")
        print(sb.driver.get_page_source())
    finally:
        sb.driver.quit()
```

Can anyone help me with that?

**Comment by mdmintz** [comment, marked as answer]
It doesn't look like `seleniumbase` was ever installed on your system.

Try:

```bash
pip install seleniumbase
```

Maybe you need something like this if you have multiple Python environments on your system:

```bash
python3 -m pip install seleniumbase
```

**Comment by renatopejon** [comment]
Doesn't these lines on the selenium Dockerfile do this?

```bash
#=====================
# Set up SeleniumBase
#=====================
COPY sbase /SeleniumBase/sbase/
COPY seleniumbase /SeleniumBase/seleniumbase/
COPY examples /SeleniumBase/examples/
COPY integrations /SeleniumBase/integrations/
COPY requirements.txt /SeleniumBase/requirements.txt
COPY setup.py /SeleniumBase/setup.py
RUN find . -name '*.pyc' -delete
RUN find . -name __pycache__ -delete
RUN pip3 install --upgrade pip setuptools wheel
RUN cd /SeleniumBase && ls && pip3 install -r requirements.txt --upgrade
RUN cd /SeleniumBase && pip3 install .
```

**Reply by mdmintz** [reply]
Yes, but that assumes the rest of the Python was set up correctly. Try using https://github.com/seleniumbase/SeleniumBase/blob/master/Dockerfile directly. Make changes as needed.

**Comment by renatopejon** [comment]
I'm using exactly this Dockerfile, I just added my this line to copy run_test_bot.sh

The script myscript.py is inside the examples folder

```bash
COPY integrations/docker/run_test_bot.sh /
```

The content of the run_test_bot.sh is:
```bash
#!/bin/bash
set -e
# Run example test from inside Docker image
echo "Running example SeleniumBase test from Docker with  Chrome..."
cd /SeleniumBase/examples/ && python myscript.py --browser=chrome
exec "$@"
```

Look at the screenshot, It works fine with the docker test, but when I try to run my script, module not found :(

![image](https://github.com/seleniumbase/SeleniumBase/assets/100096084/982efe67-a745-4649-bc7d-f38ee55f3a17)

**Reply by mdmintz** [reply]
When you ran `./run_docker_test_in_chrome.sh`, it worked. But then when you ran `./run_test_bot.sh`, it didn't. So something changed, which probably changed which Python env you were in. Note that the first time, you used `pytest` to run things, and the second time, probably not, which should help identify your issue. Also note that the included Dockerfile was specifically configured for running things via `pytest`, so if you're going to run things with pure `python`, you may need to make some changes.

**Reply by renatopejon** [reply]
Ohh I got it.

Thank you for your help, and huge congratz for this repository!

**Comment by brunobrown** [comment]
I don't know if you've already managed to solve your problem, but I had a similar import problem with another package and I managed to solve it using "-m".

Examples:

`CMD ["python", "-m", "myscript"]`

or

`CMD ["python", "-m", "utils.your_module.myscript"]`
